src/esw/can_node/debug_can_device.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+
+# For Jetson CAN info: https://docs.nvidia.com/jetson/archives/r35.2.1/DeveloperGuide/text/HR/ControllerAreaNetworkCan.html
+# Generic CAN setup for Linux:
+# After connecting to CAN, check if CAN device is connected.
+#     $ ip addr | grep "can"
+# Load the SocketCAN kernel modules
+#   CAN BUS subsystem support module:
+#     $ sudo modprobe can
+#   Raw CAN protocol module:
+#     $ sudo modprobe can_raw
+# Verify if kernel modules are loaded properly
+#     $ lsmod | grep "can"
+
+sudo apt-get install can-utils
+
+sudo modprobe can
+sudo modprobe can_raw
+
+if lsmod | grep -q "can"; then
+  ip link set can0 type can bitrate 500000 loopback on
```

Comment:
Ideally this should be by the can node... but a quick google search makes it seem like that isn't too straightforward to do.

Worst case just do `system(...)` inside of the node...

Reply:
Seems like we should actually be using https://www.netfilter.org/projects/libmnl/index.html

Reply:
Yea I was looking into it. Libmnl and netlinks seem like the more advanced way to set up CAN socket (linux/can/netlink.h exists). But I can't seem to find any good documentation anywhere.

Reply:
We can also consider https://github.com/lalten/libsocketcan/blob/gh-pages/src/libsocketcan.c

Reply:
Looks like we were able to get libnl to work. It was used to configure the bitrate and up the link. However upping the link required sudo access.

I believe under the hood this library uses the exposed Linux api for netlink. This is also what libsocketcan uses. However libnl seems more complete.